### DISCUSSION TITLE
Running application from internal memory ITCM/DTCM on MIMXRT1062

### DISCUSSION THREAD

**Discussion by asteriskSF** [zephyrproject-rtos/zephyr#69406] [Q&A] [answered]
I have a project where the board design does not include flash or external sdram.  For the final product, they want to load the application via the USB serial downloader in the ROM image and have it run solely in ITCM and DTCM.

I saw another discussion related to this, and it seems this should be possible.  https://github.com/zephyrproject-rtos/zephyr/discussions/36800#discussioncomment-6013229

I started by making a custom application and a custom board based on MIMXRT1060-EVKB.    I based the custom application off the USB console example.  I got the baseline working operating out of the NOR flash and external SDRAM. 

My next step is to switch the board configuration to  operate only in the internal ITCM/DTCM. I modified main.c to emit different text. I also modified the board**dts file:

	chosen {
		//zephyr,flash-controller = &is25wp064;
		zephyr,code-partition = &slot0_partition;
		zephyr,flash = &itcm;
		zephyr,sram = &dtcm;
		//zephyr,itcm = &itcm;
		//zephyr,dtcm = &dtcm;
		zephyr,console = &lpuart1;
		zephyr,shell-uart = &lpuart1;
		zephyr,canbus = &flexcan3;
		zephyr,display = &lcdif;
	};

The build output shows:
[7/7] Linking C executable zephyr\zephyr.elf
Memory region         Used Size  Region Size  %age Used
           SDRAM:          0 GB        32 MB      0.00%
           FLASH:       45584 B       128 KB     34.78%
             RAM:       45176 B       128 KB     34.47%
            ITCM:          0 GB       128 KB      0.00%
            DTCM:          0 GB       128 KB      0.00%
           OCRAM:          0 GB       768 KB      0.00%
        IDT_LIST:          0 GB         2 KB      0.00%

I'm hoping this indicates the dts changes were successful, but I'm not fully confident.

I confirmed with west debug that the new image does work and also that after power cycling it reverted back to the image previously loaded into external NOR flash.

My next step is to run through the process of loading the image via the USB serial debugger.
Is there a program that is compatible with NXP's serial downloaded that I can use to send the image?

**Comment by asteriskSF** [comment]
I'm also wondering how I can eliminate the CONFIG_BOOT_FLEXSPI_NOR related features from the build process.  Currently the mimxrt1060_evk/CmakeLists.txt is including:

  set(RT1060_BOARD_DIR
    "${ZEPHYR_HAL_NXP_MODULE_DIR}/mcux/mcux-sdk/boards/${BOARD_NAME}")
  if(CONFIG_BOOT_FLEXSPI_NOR)
    # Include flash configuration block for RT1060 EVK from NXP's HAL.
    # This configuration block may need modification if another flash chip is
    # used on your custom board. See NXP AN12238 for more information.
    zephyr_compile_definitions(XIP_BOOT_HEADER_ENABLE=1)
    zephyr_compile_definitions(BOARD_FLASH_SIZE=CONFIG_FLASH_SIZE*1024)
    zephyr_library_sources(${RT1060_BOARD_DIR}/xip/${FLASH_CONF})
    zephyr_library_include_directories(${RT1060_BOARD_DIR}/xip)
  endif()

None of which seems relevant if I don't have a flexspi_nor, but I don't see any option to disable CONFIG_BOOT_FLEXSPI_NOR.   
I tried:
CONFIG_BOOT_FLEXSPI_NOR=n
CONFIG_XIP=n
CONFIG_FLASH_SIZE=0
CONFIG_FLASH_BASE_ADDRESS=0

However, I run into this error:

In file included from D:/AAA/v3.5.0/zephyr/soc/arm/nxp_imx/rt/soc_rt10xx.c:15:
D:/AAA/v3.5.0/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT1062/xip/fsl_flexspi_nor_boot.h:14:10: fatal error: board.h: No such file or directory
   14 | #include "board.h"
      |          ^~~~~~~~~
compilation terminated.
[35/141] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/usb/device/usb_device.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: 'C:\Program Files\CMake\bin\cmake.EXE' --build 'D:\AAA\v3.5.0\applications\radar_app\app\build'

**Comment by DerekSnell** [comment, marked as answer]
Closing the loop here that you did move the code and data to internal SRAM, and more details are in this other discussion you started https://github.com/zephyrproject-rtos/zephyr/discussions/69630.

Best regards
